models/AmazonS3.cfc

```diff
@@ -85,13 +132,73 @@ component accessors="true" singleton {
 		buildUrlEndpoint();
 
 		// Build signature utility
-        variables.sv4Util = new Sv4Util(
-            accessKeyId        = variables.accessKey,
-            secretAccessKey    = variables.secretKey,
-            defaultRegionName  = arguments.awsRegion,
-            defaultServiceName = "s3"
-		);
+		if( variables.signature == "V4" ) {
+			variables.sv4Util = new Sv4Util(
+				accessKeyId        = variables.accessKey,
+				secretAccessKey    = variables.secretKey,
+				defaultRegionName  = arguments.awsRegion,
+				defaultServiceName = "s3"
+			);
+		} else if( variables.signature == "V2" ) {
+			variables.autoMD5 = "auto";
+		}
 
+		// manual debugging replacement for logbox
+		if( NOT structKeyExists( variables, "log" ) ) {
```

Comment:
I assume you are not using logbox, hence this code addition.

Are you also using this for your own debugging?

The concerns I have are the `request.debug` above.  We generally try to avoid checking things in the `request` scope.

Maybe this can be abstracted to a function or an `ArrayLog` component?  Just to clean this up.

The last option would be to surround all the log calls with `isNull` checks, but that doesn't seem better.

Which option sounds good to you?

Reply:
I do use/need logging, so I pulled the mini logbox into a separate component to clean it up.